### PR TITLE
Address breaking change on nightly

### PR DESCRIPTION
**Pull request by Nosferican**
Fix to [225](https://github.com/JuliaIO/JSON.jl/issues/225)

### CODE REVIEW

**Comment by TotalVerb** [comment]
Let's wait for Compat to support this so this can be done as `using Compat.Dates`.

**Comment by TotalVerb** [comment]
ref https://github.com/JuliaLang/Compat.jl/pull/413

**Comment by andreasnoack** [comment]
The new version of Compat is now in METADATA so this can be updated

File: src/Writer.jl [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module Writer
 
+using Compat
```
Comment (andreasnoack) [inline src/Writer.jl]: I think you also need a `using Compat.Dates`

Comment (Nosferican) [reply src/Writer.jl]: @nalimilan is the `using Compat.Dates` required as well?

Comment (nalimilan) [reply src/Writer.jl]: Yes, you need to replace `Base.Dates` with `Compat.Dates` everywhere it appeared. So I now realize https://github.com/JuliaData/Missings.jl/pull/56 didn't change anything to the problem.

Comment (nalimilan) [reply src/Writer.jl]: Regarding the case at hand, you just need to replace `Dates` with `Compat.Dates` on this line: https://github.com/JuliaIO/JSON.jl/blob/b4a4e338029bb999589dfabc6d90f5ae2fdd08d8/src/Writer.jl#L52

Comment (Nosferican) [reply src/Writer.jl]: Let us check change it and check Travis. After the manual fix both packages compiled so the CI should be cleared.

**Comment by stevengj** [comment]
There's no performance difference between `using Dates` and `import Dates` (followed by qualified names).  As long as there are no namespace collisions, I think `using Dates` is fine.

**Comment by TotalVerb** [comment]
Hi @Nosferican, thank you for the contribution! I went ahead and removed the `if isdefined(...)` line as mentioned in a review above. I agree that importing/using Dates is not a big deal, so this LGTM once CI passes.

**Pull request merged**